apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -544,7 +547,7 @@ func (dc *CassandraDatacenter) SetCondition(condition DatacenterCondition) {
 // GetDatacenterLabels ...
 func (dc *CassandraDatacenter) GetDatacenterLabels() map[string]string {
 	labels := dc.GetClusterLabels()
-	labels[DatacenterLabel] = CleanLabelValue(dc.Name)
+	labels[DatacenterLabel] = dc.SanitizedName()
```

Comment:
The SanitizedName() returns the wrong name. In the description it says:

``Kubernetes objects will be named after metadata.name but in Cassandra the name will be overridden by this value.``

Yet, dc.SanitizedName() returns the overridden value by default (since it calls in.CassDcName() while it should do dc.Name), not the metadata.name and thus all the objects are created with the override value.